dataqs/helpers.py

```diff
@@ -381,3 +381,18 @@ def __init__(self, content):
 
     def raise_for_status(self):
         pass
+
+
+def add_keywords(keyword_list, extra_keywords):
+    """Adds extra_keywords list to the keyword_list"""
+
+    # check if datetime and category already exist in the
+    # keyword list
+
+    keywords_added = [k for k in keyword_list
+                      if k.startswith('category:') or k.startswith('datetime:')]
+
+    if keywords_added:
+        return keyword_list[:len(extra_keywords)-1] + extra_keywords
```

Comment:
I don't think this is right.  I think you are trying to eliminate duplicate `category` and `datetime` keywords, but there is no guarantee that they will be at the end of the list.  I would just do filter out those elements (with a list comprehension like above but inverted) from the keyword list and then add the two list together at the end.